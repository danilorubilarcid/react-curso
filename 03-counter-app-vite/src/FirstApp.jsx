import PropTypes from  'prop-types';
const newMessage = 'Danilo';


const saludar = function(nombre){
    return `Hola, ${nombre}`;
}


export const PrimerComponente = ({title, subtitle, name}) => {
  
  
    return (
    <>
    <h1 data-testid="test-title">{title}</h1>
    <p>{subtitle}</p>
    <p>{name}</p>
    </>
  )
}


PrimerComponente.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string
}

PrimerComponente.defaultProps = {
    subtitle: 'subtitulo por defecto'
}