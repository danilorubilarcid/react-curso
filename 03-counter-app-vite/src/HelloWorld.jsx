/* export function App(){
    return (<h1>Hola mundo!</h1>);
}; */


import React from 'react'

export const HelloWorld = () => {
  return (
    <h1>Hola Mundo</h1>
  )
}