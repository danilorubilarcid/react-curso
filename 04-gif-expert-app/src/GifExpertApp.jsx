import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['one punch', 'dragon ball']);

    const onAddCategory = (newCategory) => {
        //Mi forma
        //setCategories(categories.concat(['Iracing']));
        //copiando arreglo
        //setCategories([...categories, 'Iracing']);
        //setCategories(cat => [...cat, 'Iracing'])
        setCategories([newCategory, ...categories]);
    }


    return(
        <>
        <h1>GifExpertApp</h1>

        <AddCategory 
        //setCategories={setCategories}
            onNewCategory={ (event) => onAddCategory(event)}
        />

        <ol>
            {
                categories.map( category => {
                    return <li key={category}>{category}</li>
                })
            }


        </ol>


        </>
    );
}
