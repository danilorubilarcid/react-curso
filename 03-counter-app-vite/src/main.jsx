import React from 'react';
import ReactDOM from 'react-dom/client';
import { PrimerComponente } from './FirstApp';
import { HelloWorld } from './HelloWorld';
import './styles.css';
import { CounterApp } from './CounterApp';


ReactDOM.createRoot (document.getElementById('root')).render(
<React.StrictMode>
    <CounterApp value={7}/>
 </React.StrictMode>

);