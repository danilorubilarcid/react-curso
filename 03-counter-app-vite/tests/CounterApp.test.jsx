import { render, screen, fireEvent } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";


describe('Pruebas al counter app copmponent', () => { 

    test('debe hacer match con el snapshot', () => { 

        const {container} = render(<CounterApp value={77}/>);

        expect(container).toMatchSnapshot();


     });

     test('el contador debe de mostrar el valkor incial de 100', () => { 
        const valor = 100;

        render(<CounterApp value={valor}/>);

        expect(screen.findByTestId('test-valor')).toBeTruthy();
        expect(screen.getByText(valor).innerHTML).toContain(valor.toString());
      });

      test('debe de incrementar con el boton +1', () => { 
        const valor = 10;

        render(<CounterApp value={valor}/>);

        fireEvent.click(screen.getByText('+1'))
        expect(screen.getByText('11')).toBeTruthy();
       })

       test('debe de decrementar con el boton -1', () => { 
        const valor = 10;

        render(<CounterApp value={valor}/>);

        fireEvent.click(screen.getByText('-1'))
        expect(screen.getByText('9')).toBeTruthy();
       })

       test('debe de resetear el valor con el boton de reset', () => { 
        const valor = 10;

        render(<CounterApp value={valor}/>);

        fireEvent.click(screen.getByText('-1'))
        expect(screen.getByText('9')).toBeTruthy();
        //fireEvent.click(screen.getByText('reset'))
        fireEvent.click(screen.getByRole('button', {name: 'btn-reset'}));
        expect(screen.getByText('10')).toBeTruthy();
       })

 })