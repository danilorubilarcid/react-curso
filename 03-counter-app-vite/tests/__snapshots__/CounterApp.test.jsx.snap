// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al counter app copmponent debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2
    data-testid="test-valor"
  >
    77
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    reset
  </button>
</div>
`;
